service/create/security_group.go

```
@@ -0,0 +1,66 @@
+package create
+
+import (
+	"fmt"
+
+	"github.com/giantswarm/awstpr"
+	"github.com/juju/errgo"
+
+	awsutil "github.com/giantswarm/aws-operator/client/aws"
+	"github.com/giantswarm/aws-operator/resources"
+	awsresources "github.com/giantswarm/aws-operator/resources/aws"
+)
+
+type securityGroupInput struct {
+	Clients           awsutil.Clients
+	Cluster           awstpr.CustomObject
+	PortsToOpen       []int
+	SecurityGroupType string
+	VpcID             string
+}
+
+func (s *Service) createSecurityGroup(input securityGroupInput) (resources.ResourceWithID, error) {
+	groupName := securityGroupName(input)
+
+	securityGroup := &awsresources.SecurityGroup{
+		Description: groupName,
+		GroupName:   groupName,
+		VpcID:       input.VpcID,
+		PortsToOpen: input.PortsToOpen,
+		AWSEntity:   awsresources.AWSEntity{Clients: input.Clients},
+	}
+	securityGroupCreated, err := securityGroup.CreateIfNotExists()
+	if err != nil {
+		s.logger.Log("error", fmt.Sprintf("could not create security group '%s': %s", groupName, errgo.Details(err)))
+		return nil, err
```

We should mask the error here